subsys/elfloader/elfloader.c

```diff
@@ -0,0 +1,363 @@
+/*
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+#include <elfloader.h>
+#include <fs/fs.h>
+#include <zephyr.h>
+#include <string.h>
+
+K_HEAP_DEFINE(_module_mem_heap, CONFIG_DYNAMIC_MODULE_MEM_SIZE * 1024);
```

Comment:
I bet anything that apps are going to want control over allocation strategy.  Probably better to have the subsystem take a k_heap or sys_heap as a parameter somewhere.